src/main/java/cms/gov/madie/measure/services/VersionService.java

```diff
@@ -279,19 +271,10 @@ public int compare(TestCase o1, TestCase o2) {
                                       .build())
                           .toList());
                 }
-                testCase =
-                    testCase.toBuilder()
-                        .id(ObjectId.get().toString())
-                        .groupPopulations(updatedTestCaseGroupPopulations)
-                        .build();
-                if (isFlagEnabled) {
-                  if (caseNumberExists) {
-                    testCase.setCaseNumber(testCase.getCaseNumber());
-                  } else {
-                    testCase.setCaseNumber(sequenceService.generateSequence(measureId));
-                  }
-                }
-                return testCase;
+                return testCase.toBuilder()
```

Comment:
I have concern with updating the input data, it is not advised to do manipulate the original data for several reasons. Can we do new TestCase() and actually clone the data? The same goes for testCaseGroupPopulation object, even though we are creating a new list updatedTestCaseGroupPopulations we are still manipulating the input data and adding it to a new List.

Reply:
@RohitKandimalla what library should I use for the clone?

Reply:
I realized that testCase.toBuilder() will anyways creates a new copy. So It can stay i guess